apps/mull-ui/src/app/pages/messages/announcements/announcements.tsx

```diff
@@ -1,16 +1,110 @@
-import React from 'react';
-import { mockPosts } from '../../../../constants';
+import { IChatForm, ISerializedPost } from '@mull/types';
+import { useFormik } from 'formik';
+import { History } from 'history';
+import React, { useContext } from 'react';
+import { Redirect } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import * as Yup from 'yup';
+import { ROUTES } from '../../../../../src/constants';
+import {
+  Event,
+  PostAddedDocument,
+  useChannelByEventQuery,
+  useCreatePostMutation,
+} from '../../../../generated/graphql';
 import { ChatInput } from '../../../components';
 import ChatBubbleList from '../../../components/chat-bubble-list/chat-bubble-list';
+import UserContext from '../../../context/user.context';
+import { useToast } from '../../../hooks/useToast';
 
-export const AnnouncementsPage = () => {
-  // TODO: Replace the following with the real announcement page implementation
+interface subscriptionData {
+  subscriptionData: {
+    data: {
+      postAdded: Partial<ISerializedPost>;
+    };
+  };
+}
 
-  return (
-    <div className="announcement-page">
-      <ChatBubbleList posts={mockPosts} />
-      <ChatInput />
-    </div>
-  );
+export interface AnnouncementPageProps {
+  history: History;
+}
+
+export const AnnouncementsPage = ({ history }: AnnouncementPageProps) => {
+  const { updateToast } = useToast();
+  const [createPostMutation] = useCreatePostMutation();
+  const { userId } = useContext(UserContext);
+  const { data, loading, error, subscribeToMore } = useChannelByEventQuery({
+    variables: {
+      eventId: 1, //TODO: Replace with dynamic event ID
+      channelName: 'Announcements',
+    },
+  });
+
+  const isEventHost = (event: Event, currentUserId: number) => {
+    return (
+      event.host.id === currentUserId || event.coHosts.some((coHost) => coHost.id === currentUserId)
+    );
+  };
+
+  const subToMore = () =>
+    subscribeToMore({
+      document: PostAddedDocument,
+      variables: { channelId: data.getChannelByEvent.id },
+      updateQuery: (prev, { subscriptionData }: subscriptionData) => {
+        if (!subscriptionData.data) return prev;
+        const newPostItem = subscriptionData.data.postAdded;
+        //Add the new message received from the subscriber
+        const newList = Object.assign({}, prev, {
+          getChannelByEvent: {
+            posts: [...prev.getChannelByEvent.posts, newPostItem],
+          },
+        });
+
+        return newList;
+      },
+    });
+
+  const formik = useFormik<IChatForm>({
+    initialValues: {
+      message: '',
+      imageFile: '',
+    },
+
+    validationSchema: Yup.object({
+      message: Yup.string().required(),
```

Comment:
Should we also consider adding a maximum length to this? Text appears to wrap for a limited amount of characters.
![image](https://user-images.githubusercontent.com/35787279/110215324-257ba400-7e77-11eb-95d3-822d99197a58.png)

Mobile: 
![image](https://user-images.githubusercontent.com/35787279/110215351-4d6b0780-7e77-11eb-9dca-992aca195e85.png)

Reply:
hmmm I think this is part of the chat bubble component, not this PR xD maybe we should open a bug report for this?

Reply:
Yeah probably should open a bug report, i didnt put it as blocking as it would be possible to address this in the user story

Reply:
Fixed! Issue was since it's a single long word it didn't want to break the word to wrap the text. I added css property `overflow-wrap: break-word`

Reply:
Yup looks good to me, my previous review still stands ✔️